### ISSUE TITLE
discontinuities at tile borders

### ISSUE THREAD

**Issue by tfriedel** [allenai/satlas#51]
In the [UI](https://satlas.allen.ai/) there are visible discontinuities / borders around tiles:
![image](https://github.com/user-attachments/assets/12767508-8bad-4029-b63d-a17043abe76a)

This is a bit of an issue for downstream use cases like segmentation (here an example with SAM2):
![image](https://github.com/user-attachments/assets/bdb19ff2-05a2-4e16-b8fd-9c87d2525c88)

Why do we see those discontinuities? I noticed in the UI there's even discontinuities for the input sentinel 2 tiles, whereas with other UIs like planetary computer I don't see those. 
How would you suggest this should be fixed?

**Comment by favyen2** [comment]
Sentinel-2 - discontinuities correspond to the border between different Sentinel-2 images. The satellites cannot image the entire planet at the same time so there are discontinuities. We use a time series of individual Sentinel-2 images as input to the model while other sources may show the pixel value averaged over many images (which is better for visualization but may be worse for the model).

Super-resolution - each tile is generated independently so there are discontinuities between tiles. There may be methods to address these discontinuities but we have not explored them.

Applying SAM on super-resolved images -- we have consistently found that super-resolution outputs may be useful for visualization, but are unhelpful as input to machine learning methods -- on every application we tested, accuracy is much much higher when inputting a sequence of Sentinel-2 images to the model. This is true as long as you have at least 50 or so training examples; if you literally have zero labels, then a pre-trained model like SAM may give better results, but it is probably very much worth it to spend time annotating 50 examples since that always boosts accuracy by a huge margin.

**Comment by tfriedel** [comment]
@favyen2 thanks for the explanation and insights!

I'm no expert on Sentinel-2, but I have looked in browsers where you can explore the imagery (e.g. earth engine, planetary explorer). And you can often select the data of the capture and then you see discontinuities between scenes, but not between tiles of a scene. So it doesn't look like the images are averaged over time if you can select them by date. I still don't quite understand why we see discontinuities in your UI between every tile, rather than scene, with the original sentinel 2 images. I believe it's probably just an issue in your UI where you apply anti-aliasing when rendering each tile. Borders between tiles are then hard cuts and stick out. 

For creating smooth super resolution results without discontinuities, maybe some approach with overlapping inputs could be used?

I agree that SAM on super resolved images is not the best approach and a finetuned U-Net style segmentation method would be better. Still the issue is that the output of this model needs to be higher resolution to properly recognize fields of size 1 ha and smaller, as is the case in countries like India. 

I saw that you got better results with low res imagery on benchmarks like PASTIS. However there are also papers that find that super resolution improves segmentation, for example in papers about medical / biological use cases. 

So maybe your results hold for countries with large fields, but not in countries with small fields? Also maybe you see worse performance because of discontinuities at tile borders?

Here's an example where I used the model from the [fields of the world](https://github.com/fieldsoftheworld/ftw-baselines) project on India and it just has not enough resolution:
![Screenshot 2024-11-09 191804](https://github.com/user-attachments/assets/d7ef65c7-c3ba-47e7-bebb-ff6e7c91ad28)

**Comment by favyen2** [comment]
The Satellite View under Super Resolution is showing an arbitrary Sentinel-2 image of the 5-18 Sentinel-2 images that the model inputs. So this would show more discontinuities. If you go to the Satellite View under any of the other four applications, it shows a monthly mosaic of Sentinel-2 images, and this should only have discontinuities between Sentinel-2 scenes. The difference is because we have only computed Super Resolution for one time range (due to cost) while the other outputs are available for a larger history.

**Comment by tfriedel** [comment]
@favyen2 
I tried your suggestion and I can still see hard borders around each tile, even if I select "Marine infrastructure" and satellite view for a specific date.
![image](https://github.com/user-attachments/assets/37f80e82-f025-4685-a066-27e2e9188e28)

[source](https://satlas.allen.ai/map?state={%22view%22:{%22longitude%22:74.12647039909643,%22latitude%22:19.020318690323577,%22zoom%22:15.627368388218056,%22pitch%22:0,%22bearing%22:0},%22showHeatmap%22:true,%22showAnimation%22:false,%22showSatellite%22:true,%22selectedDateOption%22:%222022-11%22,%22selectedDateOptionSecondary%22:%222016-01%22,%22selectedStartDateOption%22:%222016-01%22,%22selectedEndDateOption%22:%222023-12%22,%22selectedTreeCoverFill%22:[{%22label%22:%22None%22,%22isChecked%22:false},{%22label%22:%22Low%22,%22isChecked%22:false},{%22label%22:%22Medium%22,%22isChecked%22:true},{%22label%22:%22High%22,%22isChecked%22:true},{%22label%22:%22Full%22,%22isChecked%22:true}],%22selectedMarineInfra%22:[{%22label%22:%22Offshore%20Wind%20Turbines%22,%22isChecked%22:true},{%22label%22:%22Offshore%20Platforms%22,%22isChecked%22:false}],%22selectedRenewableEnergy%22:[{%22label%22:%22Solar%20Farms%22,%22isChecked%22:true},{%22label%22:%22Onshore%20Wind%20Turbines%22,%22isChecked%22:true}],%22selectedApplication%22:%22Marine%20Infrastructure%22,%22selectedLayout%22:%22Single%20Map%22})
